### PR TITLE
Add filter for results

### LINKED ISSUE #40: Add filter function to searchbar
## User stories
As user I want to filter the results of recycling search.

## Notes / further description / links
https://upmostly.com/tutorials/react-filter-filtering-arrays-in-react-with-examples

## ToDos
- [x] Add function
- [ ] Add some css styling for matching results

### PR DESCRIPTION
**Pull request by Frenzy265**
It works like I want to. But the dev tool shows me some warning: "Failed prop type: Invalid prop `value` of type `array` supplied to `InputField`, expected `string` at InputField.
I am really confused. 
I don't know, why I obviously supply an array 🤔
Maybe someone can give me a hint 😉

close #40

### CODE REVIEW

File: client/src/components/Input.js [hunk 1]
```diff
@@ -29,6 +29,6 @@ export const InputField = ({ title, value, onChange }) => {
 
 InputField.propTypes = {
   title: PropTypes.string.isRequired,
-  value: PropTypes.string.isRequired,
+  value: PropTypes.string,
```
Comment (kschmitz123) [inline client/src/components/Input.js:32]: here you say that value is a string. if you want it to be an array you need to change that

Comment (Frenzy265) [reply client/src/components/Input.js:32]: But I don't want that it is an array. The user type a word inside the inputfield, so in my mind it has to be a string.

I already try to change it to an array. But then the warning said: "Failed prop type: Invalid prop value of type `string` supplied to InputField, expected `array` at InputField.

File: client/src/pages/SearchStart.js [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 import IconArrowForward from "../assets/icon-arrow-forward-primary.svg";
 import Header from "../components/Header";
 import { useEffect, useState } from "react";
 import { Link } from "react-router-dom";
 import { getAllResults } from "../api/search";
+import { InputField } from "../components/Input";
 
 export default function Search() {
   const [results, setResults] = useState([]);
+  const [searchFilter, setSearchFilter] = useState([]);
```
Comment (kschmitz123) [inline client/src/pages/SearchStart.js]: then maybe because of this line?
```suggestion
  const [searchFilter, setSearchFilter] = useState("");
```

Comment (Frenzy265) [reply client/src/pages/SearchStart.js]: Now it works!!! 🎉  🥳   Great job! Thanks a lot!!!

**Pull request merged**